test/Chainweb/Test/Pact/RemotePactTest.hs

```diff
@@ -263,24 +259,52 @@ pollWithRetry
     :: PactTestApiCmds
     -> ClientEnv
     -> RequestKeys
-    -> IO (Either ClientError PollResponses)
+    -> IO PollResponses
 pollWithRetry cmds env rks = do
   sleep 3
   go maxPollRetries
     where
-      go retries = do
-          result <- runClientM (pollApiCmd cmds (Poll (_rkRequestKeys rks))) env
+      go !retries = do
+          let retry failure = do
+                if retries == 0
+                  then do
+                    putStrLn $ "poll failing after " ++ show maxSendRetries
+                               ++ " retries"
+                    fail failure
+                  else sleep 1 >> go (retries - 1)
+          let rkks = _rkRequestKeys rks
+          result <- runClientM (pollApiCmd cmds (Poll rkks)) env
           case result of
-              Left _ ->
-                  if retries == 0 then do
-                      putStrLn $ "poll failing after " ++ show maxSendRetries ++ " retries"
-                      return result
-                  else do
-                      sleep 1
-                      go (retries - 1)
-              Right _ -> do
-                  putStrLn $ "poll succeeded after " ++ show (maxSendRetries - retries) ++ " retries"
-                  return result
+              Left e -> retry (show e)
+              Right (PollResponses mp) -> do
+                  ok <- checkResults (toList rkks) mp
```

Comment:
can we get this to be a list of errors or something instead of the opaque boolean?

Reply:
This is test code -- and below we do dump the inputs (the result map and the input key) for comparison